src/main/java/hudson/plugins/throttleconcurrents/ThrottleQueueTaskDispatcher.java

```diff
@@ -297,6 +300,35 @@ private boolean isAnotherBuildWithSameParametersRunningOnNode(Node node, Queue.I
         return false;
     }
 
+    private Boolean isInMutualCategory(ThrottleJobProperty jobBeingChecked, ThrottleJobProperty executingJob) {
+
+        for (String category: jobBeingChecked.getCategories()) {
+            if (executingJob.getCategories().contains(category)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private Boolean isPendingAndNotFirstInLine(Queue.Item itemBeingChecked) {
```

Comment:
I am now checking if itemBeingChecked is blocked before processing the entire queue. This should help slightly. I was looking for ways to avoid iterating through the queue but getBlockedItems() (in the Queue class) isn't public. Do you have any suggestions?